### PR TITLE
Add MS Teams notification

### LINKED ISSUE #776: Ability to notify Channels on Microsoft Teams
Hello, I see that there is the ability to notify via slack.  While that is a great feature.  It would be pretty amazing if we could trigger a notification via other common messaging platforms.  Currently we use Microsoft Teams.  Where we communicate with our Developers who are using the kafka platform.

### PR DESCRIPTION
**Pull request by katsadim**
This PR resolves #776.

Desc:
The self-healing and anomaly detection is the crown feature of Cruise Control, but often the actions taken by Cruise Control provide less visibility to its operation. Adding a MS Teams notification for all the anomaly detection and self-healing will increase Cruise control visibility.

Changelog:
The MSTeamsSelfHealingNotifier extends SelfHealingNotifier and merely override the alerts method to send the notification to MS Teams.

Configuration:
The following property requires to be configured on cruisecontrol.properties

msteams.self.healing.notifier.webhook (Required)
(MSTeams webhook URL) Ref:https://docs.microsoft.com/en-us/microsoftteams/office-365-custom-connectors

To enable MSTeams notification, set `anomaly.notifier.class=com.linkedin.kafka.cruisecontrol.detector.notifier.MSTeamsSelfHealingNotifier`

Related PRs:
* https://github.com/linkedin/cruise-control/pull/629
* https://github.com/linkedin/cruise-control/pull/1510

The message looks like that:
image (4)

It would be awesome if you could label this as hacktoberfest-accepted!

### CODE REVIEW

**Review by efeg — COMMENTED** [review]
@katsadim Thanks for the PR -- left some comments.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsMessage.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
```
[the remark this answers is not in the document]
Comment (katsadim) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsMessage.java]: thanks for your review.

Addressed your comment in https://github.com/linkedin/cruise-control/pull/1713/commits/ca217b9d19443674f1f31ac9c37be906b8df2627

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifier.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
```
[the remark this answers is not in the document]
Comment (katsadim) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifier.java]: Addressed your comment in https://github.com/linkedin/cruise-control/pull/1713/commits/ca217b9d19443674f1f31ac9c37be906b8df2627

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsMessageTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
```
[the remark this answers is not in the document]
Comment (katsadim) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsMessageTest.java]: Addressed your comment in https://github.com/linkedin/cruise-control/pull/1713/commits/ca217b9d19443674f1f31ac9c37be906b8df2627

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifierTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
```
[the remark this answers is not in the document]
Comment (katsadim) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifierTest.java]: Addressed your comment in https://github.com/linkedin/cruise-control/pull/1713/commits/ca217b9d19443674f1f31ac9c37be906b8df2627

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifierTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    public void testMSTeamsAlertWithNoWebhook() {
+        _notifier = new MockMSTeamsSelfHealingNotifier(mockTime);
+        _notifier.alert(failures, false, 1L, KafkaAnomalyType.BROKER_FAILURE);
+        assertEquals(0, _notifier.getMSTeamsMessageList().size());
+    }
+
+    @Test
+    public void testMSTeamsAlertWithDefaultOptions() {
+        _notifier = new MockMSTeamsSelfHealingNotifier(mockTime);
+        _notifier._msTeamsWebhook = "http://dummy.slack.webhook";
```
[the remark this answers is not in the document]
Comment (katsadim) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/MSTeamsSelfHealingNotifierTest.java]: Addressed your comment in https://github.com/linkedin/cruise-control/pull/1713/commits/ca217b9d19443674f1f31ac9c37be906b8df2627

**Review by efeg — APPROVED** [review]
Thanks for the updates! LGTM.

**Pull request merged**